libraries/Microsoft.Bot.Builder.Dialogs/DialogContainer.cs

```diff
@@ -29,7 +29,7 @@ protected DialogContainer(string dialogId = null)
         /// Gets or sets the containers <see cref="DialogSet"/>.
         /// </summary>
         /// <value>The containers Dialog Set.</value>
-        [JsonIgnore]
+        [JsonProperty("dialogs")]
```

Comment:
Will this have no side affects? (it is not clear to me how this was previously ignored, but now is not. Should this have been a json property all along? will this affect downstream code, currently consuming the serialized version of the stack? Sorry if I'm missing something here.)

Reply:
As I test some existed bots. It didn't break any bot of them. The reason we ignore DialogSet before is we can use a method call 'EnsureDependenciesInstalled' (in AdaptiveDialog.cs)  to make sure all the referenced dialog files are loaded to DialogSet.
However, in order to support using expression in beginDialog,  the 'EnsureDependenciesInstalled' method could not resolve the dialog id which is written in expression form. So we have to use a static field to memorize the dialog ids which may be called in the expressions. Therefore, we choose to not ignore DialogSet, and added a customized deserializer to initialize the DialogSets class in DialogContainer.
As far as I know, this won't affect downstream code. And it can still deserialize the existed dialog files.